### ISSUE TITLE
`facebook/dinov2-base-imagenet1k-1-layer` is training poorly

### ISSUE THREAD

**Issue by jack89roberts** [alan-turing-institute/ARC-LoCoMoSeT#74]
E.g. See https://wandb.ai/turing-arc/locomoset/runs/ltv8zwbq?workspace=user-edable-heath - the final eval accuracy is only 30% and the loss improves then gets worse.

This is the model's HuggingFace page: https://huggingface.co/facebook/dinov2-base-imagenet1k-1-layer

Original model repo: https://github.com/facebookresearch/dinov2 (claims >80% accuracy)

Also the Renggli metric gives 0.96 for this model, e.g. see https://wandb.ai/turing-arc/locomoset/runs/lbdbz60c/overview?workspace=user-jack89roberts

**Comment by jack89roberts** [comment]
To summarise the Slack thread:

- dinov2 seems mostly designed to be used out the box and the authors didn't consider fine-tuning the whole model
- Other people seem to have had problems with fine-tuning the whole model, e.g. [see here](https://github.com/facebookresearch/dinov2/issues/276)
- It might be something about the self-supervised pre-training, though unclear why this would be the case when ultimately what we're doing is fine-tuning a pretty generic ViT architecture (the pre-trained model you get from HuggingFace)
- Some suggestions to try training the attention layers only (this seems to be motivated by [a paper](https://arxiv.org/pdf/2203.09795.pdf) that recommends it as being advantageous for fine-tuning on small datasets, but the same paper says it's less clear for larger datasets)
- [This article](https://kili-technology.com/data-labeling/computer-vision/dinov2-fine-tuning-tutorial-maximizing-accuracy-for-computer-vision-tasks#how-do-we-fine-tune-dinov2?) fine-tunes dino using two fully-connected layers as the classification head (i.e. a non-linear classification head).

**Comment by jack89roberts** [comment]
To try:

- [x] Running training for more epochs - no impact, e.g. see [this run](https://wandb.ai/turing-arc/locomoset/runs/235yv1s4?workspace=user-jack89roberts)
- [x] Changing the learning rate (smaller?) - big impact! E.g. see [this run](https://wandb.ai/turing-arc/locomoset/runs/bj1khm34/workspace?workspace=user-jack89roberts)

Other ideas:

- Training only the classification head (should ~match Renggli score)
- (maybe not necessary depending on the outcome of the above) Training a non-linear classification head only
- Training a subset of the model
